File: CiF011BackNavigationButton/app/src/main/java/ru/startandroid/cif011backnavigationbutton/MainActivity.java
package ru.startandroid.cif011backnavigationbutton;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {
    Button btnToMain2;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        btnToMain2 = (Button) findViewById(R.id.btnToMain2);
        btnToMain2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, Main2Activity.class);
                startActivity(intent);
            }
        });
    }
}

File: MaskEditTextKotlin/app/src/main/java/com/example/maskedittextkotlin/MainActivity.kt
package com.example.maskedittextkotlin

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.telephony.PhoneNumberUtils
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import com.redmadrobot.inputmask.MaskedTextChangedListener
import ru.tinkoff.decoro.MaskImpl
import ru.tinkoff.decoro.parser.PhoneNumberUnderscoreSlotsParser
import ru.tinkoff.decoro.watchers.MaskFormatWatcher

class MainActivity : AppCompatActivity() {

    internal lateinit var editText: EditText
    internal lateinit var textView: TextView
    internal lateinit var button: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        editText = findViewById(R.id.edit_text)
        textView = findViewById(R.id.textView)
        button = findViewById(R.id.button)
        button.setOnClickListener(View.OnClickListener {
            enablePhoneNumberMask(editText.text.toString())
        })

    }

    fun enablePhoneNumberMask(text: String){

        val intent = Intent(this, SecondActivity::class.java)
        startActivity(intent)



    }



}
